engine/access/rest/http/routes/collections.go

```diff
@@ -38,6 +40,9 @@ func GetCollectionByID(r *common.Request, backend access.API, link commonmodels.
 	var response commonmodels.Collection
 	err = response.Build(collection, transactions, link, r.ExpandFields)
 	if err != nil {
+		// response.Build only returns errors from the link generator (router.URLPath)
+		// which are internal configuration errors, not accessSentinel errors from the backend
+		// the ErrorHandler's catch-all will convert this to 500 Internal Server Error.
```

Comment:
are the link generator errors all internal errors, or are some related to a bad request?

Reply:
The link generator uses router.URLPath() from Gorilla Mux to generate URLs. Errors would occur if:
1. The route name doesn't exist - The route names like "getCollectionByID", "getTransactionByID" are hardcoded in the LinkGeneratorImpl methods and should match routes registered at server startup. If they don't match, it's a server misconfiguration or code bug.
2. The path variables don't match - But the IDs being passed are valid flow.Identifier values that were already successfully retrieved from the backend, so they're always in the correct format.

These errors can't be caused by user input - they only happen if:
- the server routes are misconfigured
- there's a typo in a route name
- the router isn't properly initialized